pytest_inmanta/plugin.py

```diff
@@ -951,38 +986,40 @@ def dryrun_resource(
         assert ctx.status == status
         return ctx.changes
 
-    def dryrun_all(
-        self, run_as_root: bool = False, assert_create_or_delete: bool = False
-    ) -> Dict[Resource, HandlerContext]:
+    def dryrun_all(self, run_as_root: bool = False) -> Result:
         """
         Runs a dryrun for every resource.
         :param run_as_root: run the mock agent as root
-        :param assert_create_or_delete: assert that every resource will either be created or deleted.
         """
-        results = {}
-        for resource in self.resources.values():
-            ctx = self.dryrun(resource, run_as_root=run_as_root)
-            if ctx.changes:
-                results[resource] = ctx
-            if assert_create_or_delete:
-                assert "purged" in ctx.changes
-        return results
+        return Result(
+            {
+                r: self.dryrun(r, run_as_root=run_as_root)
+                for r in self.resources.values()
+            }
+        )
 
     def dryrun_and_deploy_all(
         self, run_as_root: bool = False, assert_create_or_delete: bool = False
-    ) -> Dict[str, Dict[Resource, HandlerContext]]:
+    ) -> List[Result]:
         """
         Runs a dryrun, followed by a deploy and a final dryrun for every resource and asserts the expected behaviour.
         :param run_as_root: run the mock agent as root
         :param assert_create_or_delete: assert that every resource will either be created or deleted.
         """
-        dryrun_result = self.dryrun_all(
-            run_as_root=run_as_root, assert_create_or_delete=assert_create_or_delete
-        )
-        deploy_result = self.deploy_all(run_as_root=run_as_root)
+        first_dryrun = self.dryrun_all(run_as_root=run_as_root)
+        first_dryrun.assert_all(const.ResourceState.dry)
+
+        if assert_create_or_delete:
+            first_dryrun.assert_resources_have_purged()
+
+        deploy = self.deploy_all(run_as_root=run_as_root)
+        deploy.assert_all(const.ResourceState.deployed)
+
+        last_dryrun = self.dryrun_all(run_as_root=run_as_root)
+        first_dryrun.assert_all(const.ResourceState.dry)
+        first_dryrun.assert_has_no_changes()
```

Comment:
Is this what you meant?
```suggestion
        last_dryrun.assert_all(const.ResourceState.dry)
        last_dryrun.assert_has_no_changes()
```